src/tiled/abstractobjecttool.cpp

```diff
@@ -246,6 +247,74 @@ void AbstractObjectTool::removeObjects()
     mapDocument()->removeObjects(mapDocument()->selectedObjects());
 }
 
+/**
+ * @brief AbstractObjectTool::applyCollisionsToSelection
+ *
+ * Add the selected collision masks for the currently selected tile - the last selected tile - to
+ * all selected tiles.
+ */
+void AbstractObjectTool::applyCollisionsToSelection()
+{
+    // The selected Collision Masks
+    QList<MapObject*> selectedObjects = mapDocument()->selectedObjects();
+
+    if (auto document = DocumentManager::instance()->currentDocument()) {
+        if (auto tilesetDocument = qobject_cast<TilesetDocument*>(document)) {
+            if (Tile *currentTile = dynamic_cast<Tile *>(tilesetDocument->currentObject())) {
+
+                // Ask the user if they want to append or overwrite the collision mask
+                QMessageBox msgBox;
+                msgBox.setIcon(QMessageBox::Information);
+                msgBox.setWindowTitle(tr("Overwrite Collision Masks?"));
+                msgBox.setText(tr("Should Collision Masks be appended to, or replace, selected tiles' masks?"));
+                msgBox.addButton(tr("Append"), QMessageBox::ActionRole);
+                QPushButton *overwriteBtn = msgBox.addButton(tr("Replace"), QMessageBox::ActionRole);
+                QPushButton *abortButton = msgBox.addButton(QMessageBox::Cancel);
+                msgBox.setDefaultButton(overwriteBtn);
+                msgBox.exec();
```

Comment:
This prompt is going to get annoying fast. At the least it should only happen when any of the selected tiles already has some collision shapes. But if this action is moved to a button on the tool bar in the collision editor, then alternatively the button could also be turned into a forced menu button using `QToolButton::InstantPopup` that presents both options before triggering the action.